src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -10,28 +10,19 @@
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
-    internal class ExpressionVariableFinder : CSharpSyntaxWalker
+    internal abstract class ExpressionVariableFinder<TFieldOrLocalSymbol> : CSharpSyntaxWalker where TFieldOrLocalSymbol : Symbol
     {
-        private Binder _scopeBinder;
-        private Binder _enclosingBinder;
-        private ArrayBuilder<LocalSymbol> _localsBuilder;
+        private ArrayBuilder<TFieldOrLocalSymbol> _localsBuilder;
         private SyntaxNode _nodeToBind;
 
-        internal static void FindExpressionVariables(
-            Binder scopeBinder,
-            ArrayBuilder<LocalSymbol> builder,
-            CSharpSyntaxNode node,
-            Binder enclosingBinderOpt = null)
+        protected void FindExpressionVariables(
+            ArrayBuilder<TFieldOrLocalSymbol> builder,
+            CSharpSyntaxNode node)
         {
-            if (node == null)
-            {
-                return;
-            }
+            Debug.Assert(node != null);
 
-            var finder = s_poolInstance.Allocate();
-            finder._scopeBinder = scopeBinder;
-            finder._enclosingBinder = enclosingBinderOpt ?? scopeBinder;
-            finder._localsBuilder = builder;
+            ArrayBuilder<TFieldOrLocalSymbol> save = _localsBuilder;
+            _localsBuilder = builder;
```

Comment:
@AlekseyTs I just stumbled upon this bit. It's unclear why we're saving this field? This method is only called once.